lib/ami_spec/aws_instance.rb

```diff
@@ -21,6 +22,8 @@ def initialize(options)
       @region = options.fetch(:aws_region)
       @security_group_ids = options.fetch(:aws_security_groups)
       @tags = ec2ify_tags(options.fetch(:tags))
+      @user_data_file = options.key?(:user_data_file) ? options.fetch(:user_data_file) : nil
```

Comment:
This can be simplified to `@user_data_file = options.fetch(:user_data_file, nil)`